Mathlib/Data/Array/MinMax.lean

```diff
@@ -0,0 +1,48 @@
+/-
+Copyright (c) 2023 Thomas Murrills. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Murrills
+-/
+
+/-! # Extremizing functions on Arrays
+
+This file supplements functions like `min` and `max` on `Array`s.
+
+We introduce functions like `argmin`, which collects the elements of an array that are minimal
+under some given function.
+
+We try to use `Ord` instances where possible to stay consistent with functions like `Array.min`,
+but do also provide functions for e.g. `Preorder`s.
+
+-/
+
+-- TODO(Thomas): add the rest
+
+namespace Array
+
+section Ord
+
+variable [Ord β]
+
+/-- An auxiliary function for `argmins` which represents a step in processing the input array. It
+maintains an accumulated array of minimal elements and the minimum value encountered so far. -/
+private def argminsAux (f : α → β) : Array α × β → α → Array α × β
```

Comment:
I like to move auxiliary functions like this into `where` clauses on the main definition. I think easier to read overall, as you get to the doc-string on the "real" function first.

Reply:
Ah, me too—this was also a symptom of this being pulled from a future version of the file, which uses this function in more than one other functions :) Fixed for now in [5fbf8bd](https://github.com/leanprover-community/mathlib4/pull/2323/commits/5fbf8bde2a4d8619b867582118179ce13ab82f25).